internal/tasks/fleet_validate.go

```diff
@@ -73,8 +74,13 @@ func (t *FleetValidateLogic) CreateNewTemplateVersionIfFleetValid(ctx context.Co
 			Systemd:      fleet.Spec.Template.Spec.Systemd,
 		},
 	}
-
-	tv, err := t.store.TemplateVersion().Create(ctx, t.resourceRef.OrgID, &templateVersion, t.callbackManager.TemplateVersionCreatedCallback)
+	var callback store.TemplateVersionStoreCallback = func(m *model.TemplateVersion) {
+		t.log.Infof("fleet %v/%s: template version %s created without rollout", t.resourceRef.OrgID, t.resourceRef.Name, m.Name)
+	}
+	if fleet.Spec.RolloutPolicy == nil || fleet.Spec.RolloutPolicy.DeviceSelection == nil {
+		callback = t.callbackManager.TemplateVersionCreatedCallback
+	}
+	tv, err := t.store.TemplateVersion().Create(ctx, t.resourceRef.OrgID, &templateVersion, callback)
```

Comment:
Could we envision using a single code path / always using a rollout? In the sense of that if the user does not explicitly define a rollout policy, we default to a rollout policy that selects all devices in the fleet?

Reply:
If I understand the question correctly, does it mean that we should change the rollout flow to the new flow and not keep the old flow?
Assuming this is correct then of course we could envision that. But I would prefer the process to be gradual.  For couple of reasons:
* This would break many of the existing tests
* Users are not used to the new flow.  We might want to get some feedback before we move to the new flow completely
* This change might affect changes that are currently being worked on.  I would like to avoid as much as possible these disturbances.
* In a meeting we had with UX, We were asked specifically to keep supporting the old flow if there is no rollout definition

Reply:
The UX will remain the same, but all the code will go through the same rollout logic, as if there was a single batch and infinite disruption allowance.